wbooks/src/app/screens/Search/index.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect } from 'react';
+import { SafeAreaView, FlatList, ListRenderItem } from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+import { Book } from '@interfaces/Book';
+import actionCreators from '@redux/book/actions';
+import { AppState } from '@interfaces/Redux';
+
+import BookItem from '../BookList/components/BookItem';
+
+import styles from './styles';
+import NoItems from './components/NoItems';
+
+function Search() {
+  const dispatch = useDispatch();
+  const books = useSelector<AppState, Book[]>(state => state.bookReducer.books);
+  const searchCriteria = useSelector<AppState, string>(state => state.bookReducer.filterBooks) || '';
+
+  useEffect(() => {
+    dispatch(actionCreators.getBooks());
+  }, [dispatch]);
+
+  const keyExtractor = ({ id }: Book) => String(id);
```

Comment:
Don't repeat the same BookList implementation, split all the repeated logic and create a new component. Then you can use the same component in the Search screen and in the BookList/Library screen